### PR TITLE
Add pnpm in Node

### PR DESCRIPTION
**Pull request by akihironagai**
I have added pnpm which is a package manager to Node features.

### CODE REVIEW

File: src/node/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
-# Node.js (via nvm) and yarn (node)
+# Node.js (via nvm), yarn and pnpm (node)
```
Comment (samruddhikhandale) [inline src/node/README.md:2]: This README file is autogenerated from `devcontainer-feature.json` and will be overwritten once this PR is merged.

Can you update 👇 instead? https://github.com/devcontainers/features/blob/c8590bef3dce55bfc110fd3abdf6cd28e7d7e93f/src/node/devcontainer-feature.json#L4

File: src/node/README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
-Installs Node.js, nvm, yarn, and needed dependencies.
+Installs Node.js, nvm, yarn, pnpm, and needed dependencies.
```
Comment (samruddhikhandale) [inline src/node/README.md:4]: Same here. Can you update 👇 instead? 

https://github.com/devcontainers/features/blob/c8590bef3dce55bfc110fd3abdf6cd28e7d7e93f/src/node/devcontainer-feature.json#L6

File: src/node/install.sh [hunk 3]
```diff
@@ -103,6 +103,13 @@ else
     apt-get -y install --no-install-recommends yarn
 fi
 
+# Install pnpm
```
Comment (samruddhikhandale) [inline src/node/install.sh:106]: Can you bump the minor version? 
https://github.com/devcontainers/features/blob/c8590bef3dce55bfc110fd3abdf6cd28e7d7e93f/src/node/devcontainer-feature.json#L3

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thanks, left some comments

File: src/node/install.sh [hunk 4]
```diff
@@ -103,6 +103,13 @@ else
...
+if type pnpm > /dev/null 2>&1; then
+    echo "pnpm already installed."
+else
+    curl -fsSL https://get.pnpm.io/install.sh | sh -
```
[the remark this answers is not in the document]
Comment (akihironagai) [reply src/node/install.sh]: I avoided curling by installing pnpm from npm.

However, when using nvm and multiple versions are installed, pnpm can only be executed in first installed environment.

File: src/node/install.sh [hunk 5]
```diff
@@ -164,6 +164,13 @@ else
     fi
 fi
 
+# Install pnpm
+if type pnpm > /dev/null 2>&1; then
+    echo "pnpm already installed."
+else
+    npm install -g pnpm
```
Comment (samruddhikhandale) [inline src/node/install.sh:171]: Can you help write a test to validate these changes? Thanks! 
Adding checks to existing scenarios should be fine too - https://github.com/devcontainers/features/tree/main/test/node

File: src/node/README.md (hunk 1 above)
Comment (samruddhikhandale) [inline src/node/README.md:2]: Can we revert these changes to avoid merge conflicts?

File: src/node/install.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (samruddhikhandale) [reply src/node/install.sh]: thanks! 

> However, when using nvm and multiple versions are installed, pnpm can only be executed in first installed environment.

I think that's the way `pnpm` is currently setup. To install multiple versions of Node, one should use `pnpm` over `nvm` to make sure `pnpm` works with other versions or reinstall `pnpm` - https://pnpm.io/cli/env#use

This Feature, currently supports installing a single Node version, so I think should be fine. We can reiterate on this Feature if needed based on feedbacks.

File: src/node/README.md (hunk 1 above)
Comment (akihironagai) [reply src/node/README.md]: Reverted.

File: src/node/install.sh (hunk 5 above)
Comment (akihironagai) [reply src/node/install.sh]: I've added tests.

**Review by samruddhikhandale — APPROVED** [review]
Nice ⚡
